### ISSUE TITLE
Missing or invalid credentials. Error: connect ENOENT /run/user/1000/vscode-git-76f8f1b8ce.sock

### ISSUE THREAD

**Issue by griffi-gh** [microsoft/vscode#235523]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.95.3
- OS Version: NixOS (`nixos-unstable`)

Steps to Reproduce:

N/A, intermmittent issue

Sometimes happens during git operations that require auth, such as push/pull

Restarting or reloading the window fixes the issue

```
> git push origin master:master
Missing or invalid credentials.
Error: connect ENOENT /run/user/1000/vscode-git-76f8f1b8ce.sock
    at PipeConnectWrap.afterConnect [as oncomplete] (node:net:1607:16) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'connect',
  address: '/run/user/1000/vscode-git-76f8f1b8ce.sock'
}
Missing or invalid credentials.
Error: connect ENOENT /run/user/1000/vscode-git-76f8f1b8ce.sock
    at PipeConnectWrap.afterConnect [as oncomplete] (node:net:1607:16) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'connect',
  address: '/run/user/1000/vscode-git-76f8f1b8ce.sock'
}
remote: No anonymous write access.
fatal: Authentication failed for '<redacted>'
```

**Comment by lszomoru** [comment]
@griffi-gh, when you run into this issue could you try running the same command that is invoked by VS Code (ex:  `git push origin master:master`) from the integrated terminal and from an external terminal. Are you still hitting the same error? Thank you!

**Comment by griffi-gh** [comment]
> [@griffi-gh](https://github.com/griffi-gh), when you run into this issue could you try running the same command that is invoked by VS Code (ex: `git push origin master:master`) from the integrated terminal and from an external terminal. Are you still hitting the same error? Thank you!

Yes, the git command fails in both terminal and the ui as long as vscode's git credential helper is in use.

**Comment by lszomoru** [comment]
@griffi-gh, does it also fail in an external terminal? I am asking as there is no vscode git credential helper there...

**Comment by griffi-gh** [comment]
> [@griffi-gh](https://github.com/griffi-gh), does it also fail in an external terminal? I am asking as there is no vscode git credential helper there...

yes, the git command does work in external terminal

**Comment by lszomoru** [comment]
@griffi-gh, do you have any git credential manager configured? 

> yes, the git command does work in external terminal

Based on the description it seems that the problem is intermittent. Have you used the external terminal long enough to make sure that the problem cannot be reproduced with the external terminal?

**Comment by griffi-gh** [comment]
> 
> Based on the description it seems that the problem is intermittent. Have you used the external terminal long enough to make sure that the problem cannot be reproduced with the external terminal? 

im using the libsecret cred helper 
  
why would it happen in ext terminal if its vscode-helper-specific tho?

Btw i noticed that the Issue seems to happem more often when launching vscode using the KDE KRunner vscode workspaces addon for some reason.... (still not every time)

i ended up disabling the VSCode cred helper as a workaround

**Comment by lszomoru** [comment]
> i ended up disabling the VSCode cred helper in the settings as a workaround

Could you please elaborate on which setting you have changed? Thanks!

**Comment by griffi-gh** [comment]
> Could you please elaborate on which setting you have changed? Thanks!

```nix
"git.terminalAuthentication" = false;
"github.gitAuthentication" = false;
```

**Comment by lszomoru** [comment]
What is very interesting is that even with those settings enabled VS Code always sits at the end of the git authentication pipeline. This means that if you have a credential manager configured for git, the authentication request should never even reach VS Code. 

The only thing that I can think of at the moment is to enable [GIT_TRACE](https://git-scm.com/docs/git#Documentation/git.txt-codeGITTRACEcode) and collect a trace when the authentication fails. The trace would hopefully tell us why the authentication fails.

**Comment by rachmayadi** [comment]
Had this issue when using Rocky Linux 9. Looks like the problem lies in the terminal program. I got this error when i used the gnome-terminal. When I move to Tilix, this issue didn't occur.

**Comment by griffi-gh** [comment]
> Had this issue when using Rocky Linux 9. Looks like the problem lies in the terminal program. I got this error when i used the gnome-terminal. When I move to Tilix, this issue didn't occur.

yea, it seems to occur more or less often depending on what i use to launch vscode (the Plasma launcher extension, Konsole, or the menu entry)  
but then again, disabling vscode's builtin github auth feature is a valid workaround
